roles/icinga_agent/tasks/main.yml

```diff
@@ -10,19 +10,17 @@
   ansible.builtin.file:
     path: /etc/icinga2/repository.d
     mode: "0750"
-    owner: "{{ icinga2_user[ansible_os_family] }}"
-    group: "{{ icinga2_group[ansible_os_family] }}"
+    owner: "{{ icinga2_user[ansible_facts.os_family] }}"
+    group: "{{ icinga2_group[ansible_facts.os_family] }}"
     state: directory
-  when:
-    - icinga_agent_repositoryd_directory is defined
```

Comment:
This was removed, since it basically always evaluated to true and the directory is required by the default configuration.